### DISCUSSION TITLE
Flickering components on navigation change

### DISCUSSION THREAD

**Discussion by litemerafrukt** [remix-run/remix#3101] [Q&A]
Hi!

I am evaluating how some of the metaframworks work together with mapping libraries. Right now I'm trying Remix together with mapbbox gl wrapped in react-map-gl.

It is quite simple, a map of GOT:s Westeros with clickable kingdoms on the right and an info-panel on the left. Clicking on a kingdom marks it on the map and info-panel should fetch and present a summary of the kingdom.

The selected kingdom should be reflected in the url, eg `gotmap.se/kingdom/5`, the url is the source of truth. The thing is, whenever I navigate via `navigate` (from `useNavigate()`) the added map-sources, the kingdom-overlays, flicker. Using a store as source truth, rather than the url, works fine (no flicker) but that kinda misses the point.

I do get it working nicely with Sveltekits navigation but with Remix I am starting to pull my hair out. I found an issue on React Router [https://github.com/remix-run/react-router/issues/7634](https://github.com/remix-run/react-router/issues/7634) that might be the same issue I am having?

Is it the navigation that is the cause of flickering map-sources? Is it something I am missing here and some work around I have not found?

https://user-images.githubusercontent.com/16654263/166876227-fd86c805-292d-4524-bf2f-4264cf205e24.mp4

**Comment by lucas-giraldelli** [comment]
Did you discovered how to fix this?

**Reply by litemerafrukt** [reply]
In a way. Could not get it to work with the nice declarative apis in react-map-gl. Instead I needed to drop down to mapbox-gl directly to use featurestates and friends.

So my current hypothesis is that the problem lies more in react-map-gl than in Remix.

**Reply by lucas-giraldelli** [reply]
Oh, great answer, thanks for the insight! :)
